### PR TITLE
Improve mvi error logging

### PR DESCRIPTION
**Pull request by saneshark**
Related to - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/7080

### CODE REVIEW

**Comment by saneshark** [comment]
@kreek this is essentially ready for review. I'm going to try to add some additional specs now that the error handling and logging is done a little higher up.

File: lib/mvi/service.rb [hunk 1]
```diff
@@ -31,15 +32,46 @@ def find_profile(user)
...
     private
 
+    # TODO: Possibly consider adding Grafana Instrumentation here too
+    def mvi_error_handler(user, e)
+      case e
+      when MVI::Errors::DuplicateRecords
+        log_message_to_sentry('MVI Duplicate Record', :warn, error_context(user))
+      when MVI::Errors::RecordNotFound
+        log_message_to_sentry('MVI Record Not Found', :warn, error_context(user))
+      when MVI::Errors::InvalidRequestError
```
Comment (kreek) [inline lib/mvi/service.rb]: Are `MVI::Errors::InvalidRequestError` and `MVI::Errors::FailedRequestError` worthy of error level? An invalid request could be something that we need fix? Fully aware that I had it as warn level in the rails logger 😄

File: lib/mvi/service.rb [hunk 2]
```diff
@@ -31,15 +32,46 @@ def find_profile(user)
...
     end
 
     private
 
+    # TODO: Possibly consider adding Grafana Instrumentation here too
+    def mvi_error_handler(user, e)
+      case e
+      when MVI::Errors::DuplicateRecords
+        log_message_to_sentry('MVI Duplicate Record', :warn, error_context(user))
+      when MVI::Errors::RecordNotFound
```
Comment (kreek) [inline lib/mvi/service.rb]: I can see duplicate records being a warn but is record not found? At least for MVI I think it's expected and part the flow through the app so maybe it should be at the info level?

File: lib/mvi/service.rb (hunk 1 above)
Comment (saneshark) [reply lib/mvi/service.rb]: I'm totally open to swapping these out, lets see the complaints roll in. :-p I fully anticipate replacing this with instrumentation in grafana vs sentry, but for now that extra content is super useful (MHV launch).

File: lib/mvi/service.rb (hunk 2 above)
Comment (saneshark) [reply lib/mvi/service.rb]: Good point on that one. Record not found could become really annoying.

**Pull request merged**